### PR TITLE
Feature/76 User Auth Setup (2/2)

### PR DESCRIPTION
**Pull request by Tandid**
User authentication setup complete. 

User can't access other pages unless logged in. The only accessible pages are signup and login. This is due to the auth.global.ts middleware file that was added.

The useAuth Composable contains the authentication logic which is communicating with supabase client. This allows supabase to handle majority of the authentication.

### CODE REVIEW

File: app/composables/useAuth.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        first_name,
+        last_name,
+        email,
+        created_at: new Date(),
+        updated_at: new Date(),
+      };
+
+      const { data: insertedData, error: insertError } = await supabase
+        .from("users")
+        .insert(newUser);
```
Comment (davideastmond) [inline app/composables/useAuth.ts:39]: Can you check out the Typescript complaint on this line?

**Comment by davideastmond** [comment]
Do we need to delete the code in server/utils/database/user-db-client/user-db-client.ts ?

File: app/composables/useAuth.ts (hunk 1 above)
Comment (davideastmond) [inline app/composables/useAuth.ts:39]: Should we insert new users from here, or instead call the /users api point?

**Review by davideastmond — APPROVED** [review]

File: app/composables/useAuth.ts (hunk 1 above)
Comment (Tandid) [reply app/composables/useAuth.ts]: Just added a new commit to fix this. This also resolves the typescript complaint entirely.

Comment (Tandid) [reply app/composables/useAuth.ts]: Calling API resolved this typescript complaint

**Comment by davideastmond** [comment]
Ok let's merge

File: app/composables/useAuth.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+      if (loginError) throw loginError;
+
+      console.log("User UUID:", data.user.id); // Supabase's user auth ID
+
+      const { data: userData, error: userError } = await supabase
+        .from("users")
+        .select("*")
+        .eq("user_id", data.user.id)
+        .single();
```
Comment (davideastmond) [inline app/composables/useAuth.ts:73]: Btw I noticed this here as well where we're accessing the user table directly instead of from the API

**Pull request merged**